packages/grid/src/Grid.tsx

```diff
@@ -1,81 +1,313 @@
+          ...mappedRows.slice(0, index + 1),
+          ...newChildrens,
+          ...mappedRows.slice(index + 1),
+        ];
+        const withNewExpand = setIn(newMappedRows, true, [String(index), 'isExpand']);
+        changeMappedRows(withNewExpand);
+      }
+    },
+    [mappedRows],
+  );
+
+  const handleSelect = useCallback(
+    (e, key) => {
+      if (isDisableSelect
+           || e.target.tagName === 'svg'
+           || e.target.tagName === 'path'
+           || e.target.tagName === 'BUTTON'
+      ) return;
+      if (isMultiSelect) {
+        const newSelectedRows = addOrDeleteItemFromArray(selectedRows, key);
+        onSelect(
+          deletePropsFromObjects(
+            mappedRows.filter((el: IMappedItem) => newSelectedRows.some((id) => id === el.key)), 'key', 'expandLevel',
+          ),
+        );
+        changeSelectedRows(newSelectedRows);
+        return;
+      }
+      if (selectedRows[0] === key) {
+        onSelect({});
+        changeSelectedRows([]);
+      } else {
+        const selectedRow = {
+          ...mappedRows
+            .find((el: IMappedItem) => el.key === key),
+        } as IMappedItem;
+        delete selectedRow.key;
+        delete selectedRow.expandLevel;
+        onSelect(selectedRow);
+        changeSelectedRows([key]);
+      }
+    },
+    [isDisableSelect, isMultiSelect, selectedRows, onSelect, mappedRows],
+  );
+
+  const cell = ({
+    columnIndex, key, parent, rowIndex, style,
+  }: GridCellProps) => {
+    const content = mappedRows[rowIndex][mappedColumns[columnIndex].field];
+    const expandLevel = (!columnIndex && mappedRows[rowIndex].expandLevel) || 0;
+    const column = mappedColumns[columnIndex];
+
+    const handleExpand = () => {
+      onChangeExpand(rowIndex, mappedRows[rowIndex].children);
+    };
+
+    const isSelected = selectedRows
+      .some((k: string) => k === mappedRows[rowIndex].key);
+
+
+    return (
+      <CellMeasurer
+        cache={ cache.current }
+        columnIndex={ columnIndex }
+        key={ key }
+        parent={ parent }
+        rowIndex={ rowIndex }
+      >
+        <BodyCell
+          onClick={ (e: MouseEvent) => handleSelect(e, mappedRows[rowIndex].key) }
+          key={ key }
+          selected={ isSelected }
+          style={ {
+            ...style,
+            width: column.width,
+          } }
+          firstRow={ rowIndex === 0 }
+          even={ !!(rowIndex % 2) }
+          theme={ themeRef.current }
+        >
+          <BodyCellOffset expandLevel={ expandLevel } theme={ themeRef.current } />
+          { column.isExpandable && mappedRows[rowIndex].children ? (
+            <ExpandButtonWrapper onClick={ handleExpand }>
+              { mappedRows[rowIndex].isExpand
+                ? <RemoveIcon />
+                : <AddIcon /> }
+            </ExpandButtonWrapper>
+          ) : null }
+
+          <BodyCellContent
+            theme={ themeRef.current }
+            expandLevel={ expandLevel }
```

Comment:
Could you explain why BodyCellContent need expandLevel if BodyCellOffset responsible for it?

Reply:
to reduce the width by the indent size

Reply:
If you add `flex-shrink: 0;` to BodyCellOffset it will help.